### PR TITLE
Add categories to Japanese company provider

### PR DESCRIPTION
**Pull request by shirakia**
### What does this changes

Added variety of Japanese company names.

The category list is from Ruby Faker. https://github.com/stympy/faker/blob/master/lib/locales/ja.yml#L28

### What was wrong

Nothing was wrong.

### How this fixes it

Added categories and format patterns.

### CODE REVIEW

**Comment by shirakia** [comment]
Coveralls says coverage decreased. But this part is completely unrelated to this PR.
Can someone who has WRITE access rebuild travis? It will solve coverage decreasing.

The problem is issued here https://github.com/joke2k/faker/issues/773

File: faker/providers/company/ja_JP/__init__.py [hunk 1]
```diff
@@ -5,10 +5,15 @@
...
+        '{{last_name}}{{category}}{{company_prefix}}',
     )
 
     company_prefixes = ('株式会社', '有限会社', '合同会社')
+    categories = ('水産', '農林', '鉱業', '建設', '食品', '印刷', '電気', 'ガス', '情報', '通信', '運輸', '銀行', '保険')
 
     def company_prefix(self):
         return self.random_element(self.company_prefixes)
+
+    def category(self):
```
Comment (fcurella) [inline faker/providers/company/ja_JP/__init__.py]: Unfortunately, `category` is too generic of a word.

The problem is that in the Faker API, provider methods are not scoped by their provider, but they are all accessed from the root `Faker` instance. This means that people will call `faker.category()`, and I'm not sure we should return a list of categories that are restricted to companies.

Maybe we can just call it `company_category`?

File: faker/providers/company/ja_JP/__init__.py [hunk 2]
```diff
@@ -5,10 +5,15 @@
 
 class Provider(CompanyProvider):
     formats = (
-        '{{company_prefix}} {{last_name}}',
+        '{{company_prefix}}{{last_name}}{{category}}',
+        '{{last_name}}{{category}}{{company_prefix}}',
     )
 
     company_prefixes = ('株式会社', '有限会社', '合同会社')
+    categories = ('水産', '農林', '鉱業', '建設', '食品', '印刷', '電気', 'ガス', '情報', '通信', '運輸', '銀行', '保険')
```
Comment (fcurella) [inline faker/providers/company/ja_JP/__init__.py]: Cor easier testing, could you rename this `company_categories`?

File: faker/providers/company/ja_JP/__init__.py (hunk 1 above)
Comment (shirakia) [reply faker/providers/company/ja_JP/__init__.py]: Thanks. I understand it.
Just renamed them `company_category` and `company_categories`

File: tests/providers/test_company.py [hunk 3]
```diff
@@ -45,7 +45,7 @@ def test_company(self):
         company = self.factory.company()
         assert isinstance(company, six.string_types)
         assert any(prefix in company for prefix in prefixes)
-        assert any(company.startswith(prefix) for prefix in prefixes)
+        assert any(company.startswith(prefix) or company.endswith(prefix) for prefix in prefixes)
```
Comment (fcurella) [inline tests/providers/test_company.py]: can you also add a test that calls the new `company_category` method?

Comment (shirakia) [reply tests/providers/test_company.py]: @fcurella Added some tests 👍

**Comment by fcurella** [comment]
Thank you! ✨

**Pull request merged**